### DISCUSSION TITLE
https Proxy support to avoid port number in favor of subdomains

### DISCUSSION THREAD

**Discussion by WarDrake** [gitpod-io/openvscode-server#114] [Q&A] [answered]
I'd like to see support for subdomain proxies instead of exposing ports on the URL and server
https://vscode.example.com is much preferable than http://exmaple.com:3000
especially when running it for teams.

**Comment by akosyakov** [comment]
You should be able to install reverse proxy in front and configure it to serve on sub domain.

I don’t think it has to be built in in the server. We do the same at Gitpod.

**Reply by WarDrake** [reply]
Thanks, I created the issue before I realized the werbsocket support in the reverse proxy was the issue.

**Comment by WarDrake** [comment, marked as answer]
The answer here is to enable websockets in the reverse proxy configuration

**Reply by bhavitsharma** [reply]
Hi @WarDrake. Would you mind sharing some details how one can achieve that? I tried using nginx but looks like the community version of the k8s ingress nginx controller doesn't support websockets.

**Reply by WarDrake** [reply]
I have no experience with kubernetes ingress, but you need websocket support for this to work or you'll just get a 1056 error over and over

on nginx you get it with these lines :
```
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $http_connection;
proxy_http_version 1.1;
```
